apps/common_files/desktops/xfce.sh.erb

```diff
@@ -0,0 +1,255 @@
+    Version=1.0
+    Encoding=UTF-8
+    Name=Script
+    Type=Application
+    Exec=$AUTOSTART_SCRIPT
+    Terminal=${OOD_APP_USE_TERMINAL:-false}
+    StartupNotify=false
+    Hidden=false
+EOL
+
+    # Also make an application so you can restart it
+    realname=${OOD_APP_NAME:-OOD Autostart}
+    sed 's/^ \{4\}//' > "$XFCE_APPLICATIONS/OOD_autostart.desktop" << EOL
+    [Desktop Entry]
+    Encoding=UTF-8
+    Name=$realname
+    Name[en_GB]=$realname
+    Comment[en_GB]=(re-)start the $realname application
+    GenericName=$realname
+    GenericName[en_GB]=$realname
+
+    Exec=$AUTOSTART_SCRIPT
+    Icon=${OOD_APP_ICON:-org.xfce.session}
+    Terminal=${OOD_APP_TERMINAL:-false}
+    StartupNotify=true
+    Type=Application
+    Categories=X-Xfce-Toplevel;
+EOL
+
+fi
+
+# Create desktop path icons
+
+function make_desktop_path {
+    local path name whitespace_name
+    name="$1"
+    whitespace_name="$(echo "$name" | sed "s/_/ /g")"
+
+    path="$2"
+
+    sed 's/^ \{4\}//' > "$XFCE_APPLICATIONS/${name}.desktop" << EOL
+    [Desktop Entry]
+    Encoding=UTF-8
+    Name=$whitespace_name File Manager
+    GenericName=$whitespace_name
+
+    Exec=Thunar $path
+    Icon=org.xfce.filemanager
+    Terminal=false
+    StartupNotify=true
+    Type=Application
+    Categories=VSCFileManager;
+EOL
+}
+
+
+function make_desktop_env {
+    local name
+    name="$1"
+    if [ -n "${!name}" ]; then
+        make_desktop_path "$name" "${!name}"
+    fi
+}
+
+# # keep in sync with config vscood.rb VscOodPaths
+# if [ -d /dodrio/scratch ]; then
+#     # This is tier1
+#     PROJECTS="$HOME/.projects"
+#     if [ -e "$PROJECTS" ]; then
+#         for projdata in $(cat $PROJECTS | jq  -r '.projects[] | "\(.path):\(.slurm_account)"'); do
+#             # slurm_account as title
+#             read -r path title <<<$(echo $projdata | tr ':' ' ')
+#             make_desktop_path "project_$title" "$path"
+#         done
+#     fi
+#     for env_var in HOME VSC_HOME VSC_DATA; do
+#         make_desktop_env $env_var
+#     done
+# else
+#     for env_var in VSC_HOME VSC_DATA VSC_SCRATCH;do
+#         make_desktop_env $env_var
+#     done
+#     COURSES="/data/gent/courses/.members/$USER"
+#     if [ -e "$COURSES" ]; then
+#         for coursedata in $(cat $COURSES | jq  -r '.courses[] | "\(.path):\(.code):\(.year):\(.nickname)"'); do
+#             read -r path code year nick <<<$(echo $coursedata | tr ':' ' ')
+#             make_desktop_path "course_${nick}_${code}_${year}" "$path"
+#         done
+#     fi
+#     for env_var in VSC_DATA_VO_USER VSC_SCRATCH_VO_USER VSC_DATA_VO VSC_SCRATCH_VO VSC_SCRATCH_ARCANINE VSC_SCRATCH_ARCANINE_VO_USER VSC_SCRATCH_ARCANINE_VO; do
+#         make_desktop_env $env_var
+#     done
+# fi
+
+
+CUSTOM="$(dirname "${BASH_SOURCE[0]}")/custom.sh"
+if [ -f "$CUSTOM" ]; then
+    source $CUSTOM
+fi
+
+# Start up xfce desktop (block until user logs out of desktop)
+if [ ${SLURM_GPUS_ON_NODE:-0} -gt 0 ]; then
+    vglrun xfce4-session
```

Comment:
I though it was a bad idea to do this ?

Reply:
i agree it's better to do it for specific apps instead of the whole desktop